sa/_db-next/migrations/20171015225800_AddRenewalBit.sql

```diff
@@ -0,0 +1,14 @@
+
+-- +goose Up
+ALTER TABLE issuedNames
+       ADD COLUMN renewal TINYINT(1) NOT NULL DEFAULT 0,
+       ADD INDEX `reversedName_renewal_notBefore_Idx` (`reversedName`,`renewal`,`notBefore`),
```

Comment:
Question: Why stick `renewal` in the middle? Are we not using the `reversedName`, `notBefore` adjacent index anywhere?

Reply:
> Why stick renewal in the middle?

notBefore isn't quantized, so you'd get a list like this:

| reversedName | notBefore | renewal |
| --------------------| ---------------- | ---------|
| com.example | 2017-07-28 | true
| com.example | 2017-07-29 | false
| com.example | 2017-07-30 | true
| com.example | 2017-08-01 | true
| com.example | 2017-08-01 | false


Selecting, say, `renewal=false and > 2017-07-29`, would require scanning a bunch of rows where renewal = true, rather than using the index to find them. However, sorting them the other way gets you straight to the good stuff (columns rearranged for clarity):

| reversedName | renewal | notBefore
| --------------------| ---------------- | ---------|
| com.example | false | 2017-07-29
| com.example | false | 2017-08-01
| com.example | true | 2017-07-28
| com.example | true | 2017-07-30
| com.example | true | 2017-08-01

This is a little hard to visualize, but my rule of thumb for indexes is "put all the quantized columns first, then at most one non-quantized column."

You have a very good point about whether the existing index is used. It definitely is; I'm not sure whether the query optimizer will do the right thing with the migration. I think the right thing here is to keep both indexes in place until we land the corresponding query change to make sure we're using the index.